frontend/src/features/projects/spl/forms/GreTransferForm.tsx

```diff
@@ -0,0 +1,79 @@
+import React, { Fragment, useEffect, useMemo, useState } from 'react';
+import { ProjectDraftForm, ProjectNotes, IProject, PublicNotes, PrivateNotes } from '../../common';
+import { PropertyListViewUpdate } from '../../common/components/PropertyListViewUpdate';
+import { useFormikContext } from 'formik';
+import useCodeLookups from 'hooks/useLookupCodes';
+import _ from 'lodash';
+import { ILookupCode } from 'actions/lookupActions';
+import Form from 'react-bootstrap/Form';
+import { useSelector } from 'react-redux';
+import { RootState } from 'reducers/rootReducer';
+import { ILookupCodeState } from 'reducers/lookupCodeReducer';
+import * as API from 'constants/API';
+import { TypeaheadField } from 'components/common/form/Typeahead';
+
+/**
+ * Form component of GreTransferStep.
+ * @param param0 isReadOnly disable editing
+ */
+const GreTransferForm = ({ canEdit }: { canEdit: boolean }) => {
+  /** Enter edit mode if allowed and there are errors to display */
+  const { values, setFieldValue, touched } = useFormikContext<IProject>();
+  const agencyOptions = useCodeLookups().getOptionsByType('Agency');
+  const [initialAgencyId] = useState(values.agencyId);
+  const lookupCodes = useSelector<RootState, ILookupCode[]>(
+    state => (state.lookupCode as ILookupCodeState).lookupCodes,
+  );
+  const agencies = useMemo(
+    () =>
+      _.filter(lookupCodes, (lookupCode: ILookupCode) => {
+        return lookupCode.type === API.AGENCY_CODE_SET_NAME;
+      }),
+    [lookupCodes],
+  );
+  useEffect(() => {
+    values.properties?.forEach((property, index) => {
+      if (!isNaN(values.agencyId) && values.agencyId !== initialAgencyId) {
+        setFieldValue(`properties.${index}.agencyId`, values.agencyId);
+        const agency = _.find(agencies, { id: values.agencyId }) as ILookupCode;
+        if (agency !== undefined) {
+          setFieldValue(`properties.${index}.agencyCode`, agency.code);
+          setFieldValue(`properties.${index}.subAgency`, agency.name);
+        }
+      }
+    });
+  }, [setFieldValue, values.properties, values.agencyId, agencies, touched, initialAgencyId]);
+  return (
+    <Fragment>
+      <ProjectDraftForm isReadOnly={false} />
```

Comment:
I did a diff on this form and the original GreTransferForm and the only difference is that this line is false instead of true. Intentional?

I'd rather you pass that in as a prop, and then move the GreTransferForm to /projects/common/forms rather then 1-1 duplicating this form.

I get the desire to avoid overly complicate components with lots of conditionals to handle different uses, but if you only need to differ by a single boolean value its not really reasonable to duplicate this entire file.